File: go/problems/shuffle_string.go
package problems

// https://leetcode.com/problems/shuffle-string/
func RestoreString(s string, indices []int) string {
	res, tmp := []rune(s), []rune(s)
	for i := 0; i < len(indices); i++{
		res[indices[i]] = tmp[i]
	}
	return string(res)
}

File: rust/src/problems/shuffle_string.rs
// https://leetcode.com/problems/shuffle-string/
pub fn restore_string(s: String, indices: Vec<i32>) -> String {
  let mut ans = vec!['_'; s.len()];
  let v = s.as_bytes();
  for i in 0..indices.len() {
    ans[indices[i] as usize] = v[i] as char;
  }
  return ans.iter().collect();
}
